openmsupply/application-manager-server
# #690F fix review assignments duplicated for level 1 after LOQ is submitted

Fixes  front-end https://github.com/openmsupply/application-manager-web-app/issues/690

Finally! It took me so long to fix the tests... `generateReviewAssignments` but now all the **existing** reviewAssignments aren't returned anymore, which is what was causing the problem.

### Changes
- Update Application 4 (serial: 45678) to not be already in Stage 2 Lvl 2 - so this case can be tested in `generateReviewAssignments` to generate reviewAssignments for next level in same stage
- Fix tests in `generateReviewAssignments` and `updateReviewStatuses` to be aligned with current application/assignments
- Add test specific to show no new reviewAssignments are created if there are already some for the same: `reviewer`, `stage`, `level` and `application`.

### Testing
- Use application ABC456 logged in with user **Reviewer1**
- Start review and DECLINE one response and submit the LOQ to applicant
- Log in with user **Valerio** and fix the response then re-submit
- Log in again as **Reviewer1** to Re-review and APPROVE all then CONFORM and submit
- Now the application is on the next stage. No other reviewAssignments have been created for other reviewer in stage 1

Check in the master to see the bug happening - you can check the logs to see new reviewAssignments created after re-submitting the application with LOQ back to the reviewer.
